utils/python/CIME/user_mod_support.py

```diff
@@ -35,24 +35,24 @@ def apply_user_mods(caseroot, user_mods_path, ninst=None):
                 for comp_inst in xrange(1, ninst[comp]+1):
                     contents = newcontents
                     case_user_nl_inst = case_user_nl + "_%4.4d"%comp_inst
-                    logger.info("Pre-pending file %s"%case_user_nl_inst)
+                    logger.info("Pre-pending file %s"%(case_user_nl_inst))
                     if os.path.isfile(case_user_nl_inst):
                         with open(case_user_nl_inst, "r") as fd:
                             old_contents = fd.read()
                             if old_contents.find(contents) == -1:
                                 contents = contents + old_contents
-                    with open(case_user_nl_inst, "w") as fd:
-                        fd.write(contents)
+                                with open(case_user_nl_inst, "w") as fd:
+                                    fd.write(contents)
             else:
                 contents = newcontents
-                logger.info("Pre-pending file %s"%case_user_nl)
+                logger.info("Pre-pending file %s from %s"%(case_user_nl, include_dir))
                 if os.path.isfile(case_user_nl):
                     with open(case_user_nl, "r") as fd:
                         old_contents = fd.read()
-                        if old_contents.find(contents) == -1:
-                            contents = contents + old_contents
-                with open(case_user_nl, "w") as fd:
-                    fd.write(contents)
+                    if old_contents.find(contents) == -1:
```

Comment:
You changed the indentation of this conditional here, so that it is no longer nested under the 'with open...' - but you did not change it in the above block that applies to multi-instance. It looks to me like either one would work, though the version here (with this code _not_ nested under the 'with open...') seems better. Unless this difference was intentional, it would probably make sense to change the above block, too, so that they're consistent.